vscode/client/common/python.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/* eslint-disable @typescript-eslint/naming-convention */
+import { commands, Disposable, Event, EventEmitter, Uri } from 'vscode';
+import { PythonExtension, ResolvedEnvironment, VersionInfo } from '@vscode/python-extension';
+
+export interface IInterpreterDetails {
+    path?: string[];
+    resource?: Uri;
+}
+
+export const onDidChangePythonInterpreterEvent = new EventEmitter<IInterpreterDetails>();
+export const onDidChangePythonInterpreter: Event<IInterpreterDetails> = onDidChangePythonInterpreterEvent.event;
+
+let _api: PythonExtension | undefined;
+async function getPythonExtensionAPI(): Promise<PythonExtension | undefined> {
+    if (_api) {
+        return _api;
+    }
+    _api = await PythonExtension.api();
```

Comment:
are we sure it will never become invalid?

Reply:
We only use this Promise right after getting it and getting it each time we need it to be as close to the current configuration as we can be so there's no reason for it to be invalid in the current setup (but maybe I didn't understand the comment)